### PR TITLE
feat: home page feature added

### PR DESCRIPTION
**Pull request by ShreyashMahajan**
In this PR, 
1. Features such as categories are added. 
2. Video list data is been added.
3. Filter by category and filter by the search are added.
please review the code and provide feedback for improvement

### CODE REVIEW

**Review by ajit-sharmaGH — APPROVED** [review]
Proper namings are used, 
Code looks clean, 
You can merge it

File: src/index.js [hunk 1]
```diff
@@ -4,14 +4,23 @@ import ReactDOM from "react-dom";
 import "./index.css";
 import App from "./App";
 import { makeServer } from "./server";
+import { AuthProvider } from "./context/authContext/authContext";
+import { CategoryProvider } from "./context/categoryContext/categoryContext";
+import { VideoListProvider } from "./context/videoListContext/videoListContext";
```
Comment (SaurabhChawda) [inline src/index.js:10]: Create Index file and exports from there.

**Review by SaurabhChawda — APPROVED** [review]
Code looks fine.You can work on responsiveness of Sidebar.

**Review by cjinc121 — APPROVED** [review]
code is clean proper naming done good to merge

**Review by cjinc121 — APPROVED** [review]
.

**Pull request merged**